pkg/minishift/registration/utils.go

```diff
@@ -60,6 +61,10 @@ func RegisterHostVM(host *host.Host, param *RegistrationParameters) error {
 		if err := registrator.Register(param); err != nil {
 			return err
 		}
+		minishiftConfig.InstanceConfig.IsRegister = true
+		if err := minishiftConfig.InstanceConfig.Write(); err != nil {
```

Comment:
The actual Register and UnregisterHost methods don't have to know about our decision to persist wether a VM got registered. They should just be concerned about doing the actual registration/unregistration.